pkg/operator/configobservation/network/observe_network.go

```diff
@@ -10,57 +10,94 @@ import (
 	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
 )
 
-// ObserveRestrictedCIDRs observes list of restrictedCIDRs.
+// ObserveRestrictedCIDRs generates two configuration
 func ObserveRestrictedCIDRs(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
 	listers := genericListers.(configobservation.Listers)
 
 	var errs []error
-	restrictedCIDRsPath := []string{"admissionPluginConfig", "network.openshift.io/RestrictedEndpointsAdmission", "configuration", "restrictedCIDRs"}
+	configPath := []string{"admission", "pluginConfig", "network.openshift.io/RestrictedEndpointsAdmission", "configuration"}
 
-	previouslyObservedConfig := map[string]interface{}{}
-	if currentRestrictedCIDRBs, _, err := unstructured.NestedStringSlice(existingConfig, restrictedCIDRsPath...); len(currentRestrictedCIDRBs) > 0 {
-		if err != nil {
-			errs = append(errs, err)
-		}
-		if err := unstructured.SetNestedStringSlice(previouslyObservedConfig, currentRestrictedCIDRBs, restrictedCIDRsPath...); err != nil {
-			errs = append(errs, err)
-		}
+	admissionControllerConfig := unstructured.Unstructured{}
+	prev, ok, err := unstructured.NestedMap(existingConfig, configPath...)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if ok && err != nil {
+		// If we ever bump the API version, we'll have to do conversion here.
+		admissionControllerConfig.SetUnstructuredContent(prev)
```

Comment:
this reads oddly to me.  Why are we taking the old config?

Reply:
I was confused too. This is a straight port of the old code. It seemed to preserve previous so that it could return it in case it wasn't able to retrieve the network configuration. I'm not sure how useful that actually is.